### PR TITLE
Updated jboss-cli to latest version of æsh using low level api

### PR DESCRIPTION
**Pull request by stalep**

### CODE REVIEW

**Comment by stalep** [comment]
currently wildfly & eap6 is using a very old version of æsh that is only maintained since its used in those projects. using the latest version also adds several new functionalities we could enable if wanted. aliasing, export, etc.
it would also make it a bit easier if we in the future would be interested to use the new api in æsh.

File: pom.xml [hunk 1]
```diff
@@ -91,7 +91,7 @@
         <version.org.codehaus.woodstox.stax2-api>3.1.4</version.org.codehaus.woodstox.stax2-api>
         <version.org.codehaus.woodstox.woodstox-core-asl>4.4.1</version.org.codehaus.woodstox.woodstox-core-asl>
         <version.org.fusesource.jansi>1.9</version.org.fusesource.jansi>
-        <version.org.jboss.aesh>0.33.14</version.org.jboss.aesh>
+        <version.org.jboss.aesh>0.57-SNAPSHOT</version.org.jboss.aesh>
```
Comment (ctomc) [inline pom.xml]: don't use snapshot versions

File: cli/src/main/java/org/jboss/as/cli/impl/CliLauncher.java [hunk 2]
```diff
@@ -269,19 +272,33 @@ public static void main(String[] args) throws Exception {
...
             System.out.println(Util.getMessagesFromThrowable(t));
-            exitCode = 1;
-        } finally {
-            if((cmdCtx != null) && !gui) {
+           exitCode = 1;
+       }
+        finally {
+            if(cmdCtx != null && fileMode) {
+                //since the cli is async we give it a bit to finish
+                Thread.sleep(500);
```
Comment (ctomc) [inline cli/src/main/java/org/jboss/as/cli/impl/CliLauncher.java]: yuck

File: cli/src/main/java/org/jboss/as/cli/impl/CliLauncher.java [hunk 3]
```diff
@@ -269,19 +272,33 @@ public static void main(String[] args) throws Exception {
...
                 cmdCtx.terminateSession();
                 if(cmdCtx.getExitCode() != 0) {
                     exitCode = cmdCtx.getExitCode();
                 }
+                System.exit(exitCode);
             }
-            if (!gui) {
+            else if(cmdCtx != null && commandMode) {
+                //since the cli is async we give it a bit to finish
+                Thread.sleep(500);
```
Comment (ctomc) [inline cli/src/main/java/org/jboss/as/cli/impl/CliLauncher.java]: yuck

**Comment by stalep** [comment]
removed the sleep, something i added to test and forgot to remove before the pr.

**Comment by stalep** [comment]
the plan is to release a final of aesh 0.57 by this week, ill update the pr with the latest version.

**Comment by bstansberry** [comment]
Closing, as there is a different PR in the works.

**Pull request closed without merging**